frequency-chain/provider-dashboard
# Routes for going directly to Become a Provider

## Goal

The goal of this PR is to lay groundwork for the new Provider registration flow by setting up routes for going directly to the become-a-provider route.  

*Edit* Filed issue #194 for looking at a non-hack fix for broken tests.

Related to #184 

## Discussion
Features:
* The URL may include the name of the network to connect to. If so, it will try to connect to the given network.
* The network selection dropdown goes away (on Provider Login and Become a Provider pages) when a network is selected.  A 'Change network' link is provided which resets the component.
* The network part of the URL is optional.

So the following routes all work, and they are case-insensitive.
1. `/become-a-provider`:  which displays the network dropdown as before
2. `/become-a-provider/mainnet` : connects to mainnet immediately from the page
3. `/become-a-provider/testnet`: connects to Paseo testnet
4. `/become-a-provider/localhost` : tries to connect to localhost:9944
5. `become-a-provider/MaInNet`, etc.
6. `/mainnet`, `/testnet`, `/localhost` will each connect to the main page, select the appropriate network from the dropdown as above, and try to connect to it.
7. Custom network connections work as before.

## Checklist
- [x] routes for become-a-provider
- [x] routes for root (provider login)

## How To Test the Changes
Check out the branch and run the playwright tests.
```shell
git fetch && git co feat/new-registration-path
npx playwright test
```
Then you can play around with the new feature via `npm run dev`.

You should be able to click back and forth between Provider Login and Become a Provider pages, selecting things, then clicking Back/Cancel, and the state of the selections and connections should get reset.

You should be able to go to the URLs listed above and either connect automatically, or select from dropdown and connect, then click 'Change network' and the state should also be reset.
